services/pingpong/pingpong-base.js

```diff
@@ -0,0 +1,20 @@
+import { BaseJsonService, InvalidParameter } from '../index.js'
```

Comment:
I took the opportunity to DRY up a bit here by pulling some shared pingpong bits out to a base